kafka-junit5/src/main/java/com/salesforce/kafka/test/junit5/SharedZookeeperTestResource.java

```diff
@@ -59,4 +63,48 @@ public TestingServer getZookeeperTestServer() {
     public String getZookeeperConnectString() {
         return getZookeeperTestServer().getConnectString();
     }
+
+    /**
+     * Here we shut down the internal test zookeeper service.
+     */
+    @Override
+    public void afterAll(ExtensionContext context) throws Exception {
+        logger.info("Shutting down zookeeper test server");
+
+        // If we don't have an instance
+        if (zookeeperTestServer == null) {
+            // Nothing to close.
+            return;
+        }
+
+        try {
+            zookeeperTestServer.stop();
+            zookeeperTestServer.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        // null out reference
+        zookeeperTestServer = null;
```

Comment:
is this necessary ?

Reply:
Not entirely.  As the code stands we start the instance when we create the instance by telling the constructor to start.  We could re-use the instance and just call start on it if it exists.